### PR TITLE
Replaces all instances of the word "must" in contributing.md with "should"

### PR DESCRIPTION
**Pull request by MrsTonedOne**
Maintainers have been abusing the word "must" too much

### CODE REVIEW

**Review by LemonInTheDark — CHANGES_REQUESTED** [review]
We must not merge this pr

**Comment by Iamgoofball** [comment]
had a minor panic attack when i saw the title and immediately was relieved when i saw banned content had not been adjusted, thanks MSO, good PR

👍

**Comment by tralezab** [comment]
I honestly think that dm is an incredibly approachable language that means the bar for entry is low enough that we can enforce smarter writing for the longevity of the repository (which helps new coders anyways)

**Comment by Ghilker** [comment]
No please don't start with loosening the rules, it will lead to us becoming like bee where no rules are in place or enforced at all

**Comment by MrStonedOne** [comment]
> I honestly think that dm is an incredibly approachable language that means the bar for entry is low enough that we can enforce smarter writing for the longevity of the repository

That was how we used to do, but the most recent crop of maintainers critically lack critical thinking skills. every maintainer added after rohesie (including rohesie) has treated this file (and any sort of programming trend they thought up but hadn't yet put into this file) as a sort of bludgeoning stick to apply to anything resembling the trend they decided to hate, without regard to the objectives behind the contributing.md addition.

Since maintainers can't be trusted to use logic and an individualized review to determine if it makes sense for each individual case to have the standard applied to it, this is the alternative.

**Comment by Iamgoofball** [comment]
i dunno about that one bro, i kinda like that the maintainers dont let me shitcode all day

File: .github/CONTRIBUTING.md
Comment (Mothblocks) [inline .github/CONTRIBUTING.md:815]: 😄

**Comment by Mothblocks** [comment]
like more than half the stuff in here we either lint for or rely upon for other reasons (such as SIGNAL_HANDLER) lol

**Review by Mothblocks — CHANGES_REQUESTED** [review]
image

**Comment by LemonInTheDark** [comment]
> That was how we used to do, but the most recent crop of maintainers critically lack critical thinking skills. every maintainer added after rohesie (including rohesie) has treated this file (and any sort of programming trend they thought up but hadn't yet put into this file) as a sort of bludgeoning stick to apply to anything resembling the trend they decided to hate, without regard to the objectives behind the contributing.md addition.

This isn't totally the case. We might have different standards, but there have been prs in the vein that you describe that were closed because we agreed that standardizing things that don't work as standards is pointless. See https://github.com/tgstation/tgstation/pull/57589#issuecomment-796109374

> Since maintainers can't be trusted to use logic and an individualized review to determine if it makes sense for each individual case to have the standard applied to it, this is the alternative.

Much of what you're touching here very much should remain as "must". It would be better if you made prs that removed the changes you dislike for the above reasoning, so things can be discussed there. Hell I think only like 2 or three of these lines even come post rohesie. Could be wrong about that, since I am mostly post rohesie

File: .github/CONTRIBUTING.md
Comment (Thunder12345) [inline .github/CONTRIBUTING.md:96]: I doubt any non object-oriented code would get merged regardless of what this says

File: .github/CONTRIBUTING.md
Comment (Thunder12345) [inline .github/CONTRIBUTING.md:98]: Relative pathing is the work of the devil and should never be allowed under any circumstances

File: .github/CONTRIBUTING.md
Comment (Thunder12345) [inline .github/CONTRIBUTING.md:129]: Same here, don't encourage this type of shitcode

File: .github/CONTRIBUTING.md
Comment (Thunder12345) [inline .github/CONTRIBUTING.md:187]: Coding standards should not be optional, there is an easy way to do this properly

File: .github/CONTRIBUTING.md
Comment (Thunder12345) [inline .github/CONTRIBUTING.md:189]: Coding standards should not be optional

File: .github/CONTRIBUTING.md
Comment (Thunder12345) [inline .github/CONTRIBUTING.md:192]: Coding standards should not be optional

File: .github/CONTRIBUTING.md
Comment (Thunder12345) [inline .github/CONTRIBUTING.md:195]: Coding standards should not be optional

File: .github/CONTRIBUTING.md
Comment (Thunder12345) [inline .github/CONTRIBUTING.md:213]: Mixing tabs and spaces requires immediate banishment to the depths of hell, do not encourage this heathen behaviour

File: .github/CONTRIBUTING.md
Comment (Thunder12345) [inline .github/CONTRIBUTING.md:230]: The quote no longer makes sense with "should"

File: .github/CONTRIBUTING.md
Comment (Thunder12345) [inline .github/CONTRIBUTING.md:399]: This is a security issue, it must be done properly

File: .github/CONTRIBUTING.md
Comment (Thunder12345) [inline .github/CONTRIBUTING.md:401]: This is a security issue, it must be done properly

File: .github/CONTRIBUTING.md
Comment (Thunder12345) [inline .github/CONTRIBUTING.md:403]: This is a security issue, it must be done properly

File: .github/CONTRIBUTING.md
Comment (Thunder12345) [inline .github/CONTRIBUTING.md:421]: How to break a database in one easy step

File: .github/CONTRIBUTING.md
Comment (Thunder12345) [inline .github/CONTRIBUTING.md:423]: How to break a database in one easy step version two

File: .github/CONTRIBUTING.md
Comment (Thunder12345) [inline .github/CONTRIBUTING.md:425]: Bad practice to hardcode this, don't encourage it

File: .github/CONTRIBUTING.md
Comment (Thunder12345) [inline .github/CONTRIBUTING.md:427]: How to break a database in one easy step version three

File: .github/CONTRIBUTING.md
Comment (Thunder12345) [inline .github/CONTRIBUTING.md:429]: I don't know what this means but it's probably best not to water down code standards without a good justification

File: .github/CONTRIBUTING.md
Comment (Thunder12345) [inline .github/CONTRIBUTING.md:437]: Leaving the door open to disappointment, they should really have already been in discussions with a maptainer much earlier in the process than even this

File: .github/CONTRIBUTING.md
Comment (Thunder12345) [inline .github/CONTRIBUTING.md:440]: TGM should always be used, it is better at handling conflicts and functions exactly the same in the eyes of BYOND. There is no reason to ever not use this

File: .github/CONTRIBUTING.md
Comment (Thunder12345) [inline .github/CONTRIBUTING.md:451]: This could be said to depend on one's opinion of TGUI, again needs a justification before watering down code standards

File: .github/CONTRIBUTING.md
Comment (Thunder12345) [inline .github/CONTRIBUTING.md:458]: It must, or it won't work properly

File: .github/CONTRIBUTING.md
Comment (Thunder12345) [inline .github/CONTRIBUTING.md:466]: It must, or it won't work properly

File: .github/CONTRIBUTING.md
Comment (Thunder12345) [inline .github/CONTRIBUTING.md:529]: Coding standards should not be optional

File: .github/CONTRIBUTING.md
Comment (Thunder12345) [inline .github/CONTRIBUTING.md:704]: This is good practice and it should not be encouraged to avoid doing it

File: .github/CONTRIBUTING.md
Comment (Mothblocks) [inline .github/CONTRIBUTING.md:399]: 🤔

File: .github/CONTRIBUTING.md
Comment (Thunder12345) [inline .github/CONTRIBUTING.md:737]: It must, or it won't work properly

File: .github/CONTRIBUTING.md
Comment (Thunder12345) [inline .github/CONTRIBUTING.md:795]: Well if you don't profile it how do you know it's faster?

File: .github/CONTRIBUTING.md
Comment (Thunder12345) [inline .github/CONTRIBUTING.md:813]: Lol

File: .github/CONTRIBUTING.md
Comment (Thunder12345) [inline .github/CONTRIBUTING.md:815]: Lol

File: .github/CONTRIBUTING.md
Comment (Mothblocks) [reply .github/CONTRIBUTING.md:98]: its also linted against

File: .github/CONTRIBUTING.md
Comment (Mothblocks) [reply .github/CONTRIBUTING.md:189]: this is also linted against

File: .github/CONTRIBUTING.md
Comment (Mothblocks) [reply .github/CONTRIBUTING.md:213]: this is also linted against

File: .github/CONTRIBUTING.md
Comment (interestingusernam3) [inline .github/CONTRIBUTING.md:815]: lol

File: .github/CONTRIBUTING.md
Comment (MrStonedOne) [reply .github/CONTRIBUTING.md:429]: While this entire pr was made from a 5 second find/replace, this line here I added in, and I support moving to should.

File: .github/CONTRIBUTING.md
Comment (Kylerace) [reply .github/CONTRIBUTING.md:737]: not true, this is because it WILL work, dm assumes that if its not a define or a var that what you are declaring as a associative key must be a string, so if you meant to make it a var or define but you misspell it then byond will not tell you
Screenshot_1085

**Comment by optimumtact** [comment]
> > I honestly think that dm is an incredibly approachable language that means the bar for entry is low enough that we can enforce smarter writing for the longevity of the repository
> 
> That was how we used to do, but the most recent crop of maintainers critically lack critical thinking skills. every maintainer added after rohesie (including rohesie) has treated this file (and any sort of programming trend they thought up but hadn't yet put into this file) as a sort of bludgeoning stick to apply to anything resembling the trend they decided to hate, without regard to the objectives behind the contributing.md addition.
> 
> Since maintainers can't be trusted to use logic and an individualized review to determine if it makes sense for each individual case to have the standard applied to it, this is the alternative.

giphy

File: .github/CONTRIBUTING.md
Comment (LemonInTheDark) [reply .github/CONTRIBUTING.md:737]: if you use this to remove gaslist typepaths I will rain hellfire on you

**Pull request closed without merging**
